spatial_id_detector.go

```diff
@@ -0,0 +1,68 @@
+package spatialID
+
+import (
+	radixtree "github.com/trajectoryjp/multidimensional-radix-tree/src/tree"
+)
+
+type SpatialIDDetector interface {
+	IsOverlap(ids SpatialIDs) bool
+}
+
+type SpatialIDGreedyDetector struct {
+	ids SpatialIDs
+}
+
+func NewSpatialIDGreedyDetector(ids SpatialIDs) SpatialIDDetector {
+	return &SpatialIDGreedyDetector{ids}
+}
+
+func (detector *SpatialIDGreedyDetector) IsOverlap(targetIDs SpatialIDs) bool {
+	return detector.ids.Overlaps(targetIDs)
+}
+
+type SpatialIDTreeDetector struct {
```

Comment:
In the future we plan to implement merging etc on this.
So rename this to `SpatialIDTree` for extended functionality.